src/utils/localDockerDatabase.ts

```diff
@@ -0,0 +1,224 @@
+import Docker, { Container } from "dockerode";
+import { printAdaptiveLog } from "./logging.js";
+import log from "loglevel";
+import { YamlDatabaseConfiguration } from "../models/yamlProjectConfiguration.js";
+import { getProjectConfiguration } from "./configuration.js";
+
+export async function runDockerDatabaseContainer(
+    dockerHandler: Docker,
+    databaseType: string,
+    projectName: string,
+) {
+    const containerName = `/genezio-${databaseType}-${projectName}`;
+    let currentContainer: Container | undefined;
+    try {
+        const containers = await dockerHandler.listContainers({ all: true });
+        containers.forEach(function (container) {
+            if (container.Names[0] == containerName) {
+                currentContainer = dockerHandler.getContainer(container.Id);
+            }
+        });
+    } catch (error) {
+        printAdaptiveLog(`Failed starting local docker ${databaseType} database`, "");
+        if (error instanceof Error) {
+            log.error(`An error has occured: ${error.toString()}`);
+        }
+        log.error(
+            "This is most likely a docker error, check if your docker daemon is running.\nIf it is not running you can learn how to start it by going to the docker documentation at https://docs.docker.com/ ",
+        );
+        return undefined;
+    }
+    if (currentContainer) {
+        const inspectRes = await currentContainer.inspect();
+        const isContainerRunning = inspectRes.State.Running;
+        if (!isContainerRunning) {
+            try {
+                await currentContainer.start();
+            } catch (error) {
+                printAdaptiveLog(`Failed starting local docker ${databaseType} database`, "");
+                if (error instanceof Error) {
+                    log.error(`An error has occured: ${error.toString()}`);
+                }
+                return undefined;
+            }
+        }
+    } else {
+        try {
+            switch (databaseType) {
+                case "postgres": {
+                    let isImagePresent = false;
+                    const images = await dockerHandler.listImages({ all: true });
+                    images.forEach(function (image) {
+                        if (image.RepoTags && image.RepoTags[0] == "postgres:latest") {
+                            isImagePresent = true;
+                        }
+                    });
+                    if (!isImagePresent) {
+                        const pullStream = await dockerHandler.pull("postgres");
+                        await new Promise((res) =>
+                            dockerHandler.modem.followProgress(pullStream, res),
+                        );
+                    }
+                    const newContainer = await dockerHandler.createContainer({
+                        Image: "postgres",
+                        Env: [`POSTGRES_PASSWORD=mysecretpassword`],
```

Comment:
I was thinking that the container should inherit the local process ENV. I am not sure about this, you don't need to change anything for now, just keep in mind that maybe we should think about this and change it in the future.